src/Gui/Hooks/useStoryPreview/init.luau

```diff
@@ -0,0 +1,27 @@
+local Hoarcekat = script:FindFirstAncestor("Hoarcekat")
+
+local React = require(Hoarcekat.Packages.React)
+
+local StoryPreviewer = require(script.StoryPreviewer)
+
+local function useStoryPreview(rootRef, selectedStory, expand, setCurrentPreview)
+	local storyPreviewer = React.useRef(StoryPreviewer.new(rootRef, setCurrentPreview))
```

Comment:
This will call `StoryPreviewer.new` and by extension create new screengui instances every render. You may want to memo this instead.

Reply:
Better alternative since memo is for optimizations only - keep the ref and initially assign it by nil checking during render.